check_type.py

```diff
@@ -0,0 +1,40 @@
+from functools import wraps
+
+
+def decorator_factory(correct_type: type):
+    """
+    A decorator that gets an object type
+    and returns a decorator that checks if the variable type is the type of object that the function gets.
+    :param correct_type: The correct object type.
+    :return: decorator function
+    """
+    def decorator(func: callable):
+        """
+        A decorator that checks whether the function gets the variable of the correct type.
+        If it is incorrect, an error message is sent.
+        :param func:The function that receives an object.
+        :return:The function it gets.
+        """
+        @wraps(func)
+        def type_check(obj: object):
+            """
+        Checks whether the function receives the correct type object  according to the decorator if no error is sent.
+            :param obj: Object to test
```

Comment:
Don't use abbreviations in variable names -
https://softwareengineering.stackexchange.com/questions/72593/how-to-abbreviate-variable-names